Sources/PowerAuth/PowerAuthError.swift

```diff
@@ -0,0 +1,202 @@
+    
+    // Configuration & State
+    
+    /// The underlying reason the `.invalidConfiguration` error occured.
+    public enum ConfigurationFailureReason {
+        /// `PowerAuthConfiguration` contains missing or invalid data.
+        case invalidInstanceConfiguration
+        
+        /// `KeychainConfiguration` contains missing or invalid data.
+        case invalidKeychainConfiguration
+        
+        /// `HttpClientConfiguration` contains missing or invalid data.
+        case invalidHttpClientConfiguration
+    }
+    
+    /// The provided configuration is not valid. You can check the debug log for more details.
+    case invalidConfiguration(reason: ConfigurationFailureReason)
+    
+    /// The provided parameter to function is not valid. Please check the debug log for more details.
+    case invalidParameter
+    
+    
+    /// The underlying reason the `.invalidActivationState` error occured.
+    public enum ActivationStateFailureReason {
+        
+        /// `PowerAuth` instance has already a valid activation.
+        case activationIsPresent
+        
+        /// `PowerAuth` instance has already pending an activation process.
+        case pendingActivation
+        
+        /// `PowerAuth` instance has no activation.
+        case missingActivation
+        
+        /// `PowerAuth` instance already has a biometry factor key configured.
+        case biometryFactorAlreadySet
+        
+        /// `PowerAuth` instance doesn't contain an activation recovery data.
+        case missingActivationRecoveryData
+        
+        /// TODO: Temporary reason for errors reported from PowerAuthCore.
+        ///       We should enhance error codes from core to better match wrong states.
+        case other
+    }
+    
+    /// The operation was requested in wrong `PowerAuth` activation state.
+    case invalidActivationState(reason: ActivationStateFailureReason)
+    
+    
+    // Activation
+    
+    /// The underlying reason the `.invalidActivationData` error occured.
+    public enum ActivationDataFailureReason {
+        /// You have provided wrong activation code.
+        case wrongActivationCode
+        /// The signature scanned from QR code with activation code doesn't match.
+        case wrongActivationSignature
+        /// You have provided invalid recovery code.
+        case wrongRecoveryCode
+        /// You have provided invalid recovery PUK.
+        case wrongRecoveryPuk
+        /// Provided additional activation OTP contains an empty string.
+        case emptyOtp
+        /// Identity attributes provided for a custom attributes is empty.
+        case emptyIdentityAttributes
+    }
+    
+    /// Data provided to `Activation` structure are invalid.
+    case invalidActivationData(reason: ActivationDataFailureReason)
+    
+    
+    // Authentication
+    
+    /// The underlying reason the `.invalidAuthenticationData` error occured.
+    public enum AuthenticationDataFailureReason {
```

Comment:
Why the `Is` verbs in the cases? Nowhere else in this file are used enum cases with verbs.